src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4951,7 +5013,7 @@ private bool IsCurrentTokenPartialKeywordOfPartialMethodOrType()
         {
             if (this.CurrentToken.ContextualKind == SyntaxKind.PartialKeyword)
             {
-                if (this.IsPartialType() || this.IsPartialMember())
+                if (ShouldPartialBeTreatedAsModifier() != PartialModFlags.NotModifier)
```

Comment:
Would read better as ```if (AnalyzePartialUsafe() ...```

Reply:
whole method could become:

```c#
return this.CurrentToken.ContextualKind == SyntaxKind.PartialKeyword &&
       ShouldPartialBeTreatedAsModifier() != PartialModFlags.NotModifier;
```

Reply:
I tend to not reformat code anywhere for better diff and easier review.